### DISCUSSION TITLE
How do you do a clean dev build?

### DISCUSSION THREAD

**Discussion by gvdevx** [immich-app/immich#4093] [Q&A] [answered]
I grabbed the code and did 'make dev' on my computer. I'm running Windows 11 with WSL2.

I did the above in an Ubuntu shell. It created all the docker images and ran. Great! But, I have a problem with nginx in that the two shell scripts have Windows style line endings so the bash scripts fail to run in the docker image.

I have edited the files locally and fixed up the line endings. However whenever I do 'make dev' it never grabs the new files. I can of course do a docker copy and replace the files in the image. But I would rather do a clean and start the build fresh (maybe I have other changes too).

How do I do this?

**Comment by alextran1502** [comment, marked as answer]
You can stop and remove the container with `docker-compose down` and then `docker container prune`

If you want to clean the volumes, check the volume name with `docker volume ls` and then `docker volume rm <volume-name>`

If you change any dependency, you can run `make dev-update` to install any new dependencies in the containers. 

I hope those information help

**Comment by gvdevx** [comment]
Ah thanks. Yeah I did all the deleting of images but that didn't help as it seemed to somehow get old data from somewhere. I remember the first time through it would compile code, install scripts (took forever). But this is what I want, for it to compile from scratch again. I have never been able to get back to that. I'm confused as to where it is finding this pre-built image stuff.

I will try the make dev-update and see if that gets me anywhere.

I am used to programming environments where you delete .obj .bin files etc and everything compiles fresh. I am new to this docker way of  doing things. Thanks for your quick reply.

**Reply by alextran1502** [reply]
Docker has persistent volume, which we use for Typesense and Postgres database, you will need to remove them to start fresh

**Comment by jrasm91** [comment]
You should configure git to checkout files "as is" btw, and not to use CRLF.

**Comment by gvdevx** [comment]
Yes, that one mistake has caused me a world of pain.
